source_middleware.go

```diff
@@ -309,12 +326,12 @@ func (s *sourceWithSchemaExtraction) extractPayloadSchema(ctx context.Context, r
 }
 
 func (s *sourceWithSchemaExtraction) schemaForType(ctx context.Context, data any, subject string) (schema.Schema, error) {
-	srd, err := schema.KnownSerdeFactories[s.config.SchemaType].SerdeForType(data)
+	srd, err := schema.KnownSerdeFactories[s.config.SchemaType()].SerdeForType(data)
```

Comment:
I think we should check whether `s.config.SchemaType()` returns a valid known type.

Reply:
We rely on that `s.config.SchemaType()` converts `s.config.SchemaTypeStr` to a `schema.Type`, and `SchemaTypeStr` has already been validated. But I think you have a point, I can add a test or two to make sure `s.config.SchemaType()` is doing the conversion successfully.

Reply:
@hariso Just checking in. You haven't added a test for this, did you? Just in case I'm not seeing it.

Reply:
Ah, good catch, I forgot about this comment.:) I'll add a test or two now.